src/transaction/log_page_buffer.c

```diff
@@ -1997,12 +2002,22 @@ logpb_flush_header (THREAD_ENTRY * thread_p)
  *              If not, read log page from log.
  */
 LOG_PAGE *
-logpb_fetch_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, LOG_PAGE * log_pgptr)
+logpb_fetch_page (THREAD_ENTRY * thread_p, LOG_LSA * req_lsa, LOG_CS_ACCESS_MODE access_mode, LOG_PAGE * log_pgptr)
 {
   LOG_PAGE *ret_pgptr = NULL;
+  LOG_LSA append_lsa, append_prev_lsa;
 
   assert (log_pgptr != NULL);
-  assert (pageid != NULL_PAGEID);
+  assert (req_lsa != NULL);
+  assert (req_lsa->pageid != NULL_PAGEID);
+
+  if (access_mode != LOG_CS_SAFE_READER && VACUUM_IS_PROCESS_LOG_FOR_VACUUM (thread_p))
+    {
+      access_mode = LOG_CS_SAFE_READER;
```

Comment:
The code was added in context of using macros (from vacuum) such as LOG_READ_ADVANCE_WHEN_DOESNT_FIT, where the used mode is LOG_CS_FORCE_USE.
The code overrides the mode to LOG_CS_SAFE_READER.